src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/NetServer.java

```diff
@@ -234,5 +246,14 @@ public void stop() {
     // Remove handlers.
     sendDataHandler = null;
     subscribeHandler = null;
+
+    if (server != null) {
+      server.shutdown();
+      try {
+        server.awaitTermination(1, TimeUnit.MINUTES);
+      } catch (InterruptedException e) {
+        server.shutdownNow();
+      }
+    }
```

Comment:
Are you sure you tested this change? We don't want to stop the server, it'll kill perftop and the ability to query metrics as well. If you want to change the behavior, I'd suggest we create a new issue and address this the right way in a new PR.

For the short term, you can consider renaming the method to removeRcaHandlers() or something like that.

Reply:
This is inside the stop method. I'm surprised that the stop method doesn't actually stop the server. Just in case we're relying on stop not to stop the server, I've removed the logic from stop() and added a shutdown() method which actually shuts everything down.